focuspoints.lrdevplugin/ExifUtils.lua

```diff
@@ -42,37 +42,44 @@ function ExifUtils.readMetaData(targetPhoto)
   return fileInfo
 end
 
-function ExifUtils.filterInput(str)
-  --local result = string.gsub(str, "[^a-zA-Z0-9 ,\\./;'\\<>\\?:\\\"\\{\\}\\|!@#\\$%\\^\\&\\*\\(\\)_\\+\\=-\\[\\]~`]", "?");
-  -- FIXME: doesn't strip - or ] correctly
-  local result = string.gsub(str, "[^a-zA-Z0-9 ,\\./;'\\<>\\?:\\\"\\{\\}\\|!@#\\$%\\^\\&\\*\\(\\)_\\+\\=\\-\\[\\\n\\\t~`-]", "?");
-  return result
-end
+--[[
+-- Transforms the output of ExifUtils.readMetaData and returns a key/value lua Table
+-- targetPhoto - LrPhoto to extract the Exif from
+--]]
+function ExifUtils.readMetaDataAsTable(targetPhoto)
+  local metaData = ExifUtils.readMetaData(targetPhoto)
+  if metaData == nil then
+    return nil
+  end
 
-function ExifUtils.findValue(metaData, key)
-  local parts = ExifUtils.createParts(metaData)
-  local labels = ""
-  local values = ""
-  for k in pairs(parts) do
-    local l = parts[k].key
-    local v = parts[k].value
-    if (l == nill) then l = "" end
-    if (v == nill) then v = "" end
-    l = LrStringUtils.trimWhitespace(l)
-    v = LrStringUtils.trimWhitespace(v)
-    if (key == l) then
-      return v
-    end
+  local parsedTable = {}
+
+  for keyword, value in string.gmatch(metaData, "([^\:]+)\:([^\r\n]*)\r?\n") do
+    keyword = LrStringUtils.trimWhitespace(keyword)
+    value = LrStringUtils.trimWhitespace(value)
+    parsedTable[keyword] = value
+    log("EXIF | Parsed '" .. keyword .. "' = '" .. value .. "'")
   end
 
-  return nil
+  return parsedTable
 end
 
-function ExifUtils.findFirstMatchingValue(metaData, keys)
+--[[
+-- Returns the first value of "keys" that could be found within the metaDataTable table
+-- Ignores nil and "(none)" as valid values
+-- metaDataTable - the medaData key/value table
+-- keys - the keys to be search for in order of importance
+--]]
+function ExifUtils.findFirstMatchingValue(metaDataTable, keys)
   local value = nil
 
-  for key,keyword in pairs(keys) do
-    value = ExifUtils.findValue(metaData, keyword)
+  for key, keyword in pairs(keys) do
```

Comment:
nit: 
this is typically a "key -> value" loop similar to a HashMap. Mind calling k,v or key, value (and renaming the prop on line 74)?